scripts/zkai_attention_kv_two_head_fused_softmax_table_native_gate.py

```
@@ -0,0 +1,1050 @@
+def validate_source_artifacts(
+    source_input: dict[str, Any],
+    source_envelope: dict[str, Any],
+    sidecar_envelope: dict[str, Any],
+    *,
+    source_envelope_bytes: bytes,
+    sidecar_envelope_bytes: bytes,
+) -> None:
+    try:
+        SOURCE_INPUT_MODULE.validate_payload(source_input)
+    except Exception as err:
+        raise AttentionKvTwoHeadFusedSoftmaxTableGateError(f"source input validation drift: {err}") from err
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)